src/target/adiv5_swd.c

```diff
@@ -234,7 +235,7 @@ bool adiv5_swd_scan(const uint32_t targetid)
 		bool scan_multidrop = targetid || dp->version >= 2U;
 
 #if PC_HOSTED == 1
-	if (scan_multidrop && !dp->dp_low_write) {
+	if (scan_multidrop && !dp->write_no_check) {
 		DEBUG_WARN("Discovered multi-drop enabled target but CMSIS_DAP < v1.2 cannot handle multi-drop\n");
 		scan_multidrop = false;
```

Comment:
Just realising that this might need to mark the DP as being "v1" only to inhibit the TARGETSEL logic in error recovery as that's what this is really all about iwrc.. need to very carefully think on how this code interacts with protocol and error recovery.

Reply:
This particular detail has been iching me for a while, but I never dove into it, so I dont know what goes into it

Reply:
I did some reading, looks like the command used for the sequence in out was only [added in v1.2.0](https://www.keil.com/pack/doc/CMSIS/DAP/html/dap_revisionHistory.html), the command is [DAP_SWD_Sequence](https://www.keil.com/pack/doc/CMSIS/DAP/html/group__DAP__SWD__Sequence.html) and critically, it allows for a input sequence (SWDIO as input), which the previous existing command didn't ([DAP_SWJ_Sequence](https://www.keil.com/pack/doc/CMSIS/DAP/html/group__DAP__SWJ__Sequence.html)), this is required for the target select, which does indeed affect the protocol recovery along with the multi-drop. 

What it looks like to me as that the current implementation is wholesale broken for v < 1.2, as it generally makes use of the sequence command which isn't available, I may be missing some parts though.

As an extra note, I do wonder if its possible to replicate a input sequence with a sequence of [DAP_SWJ_Pins](https://www.keil.com/pack/doc/CMSIS/DAP/html/group__DAP__SWJ__Pins.html), the documentation does not make the behaviour clear (when are the pins configured as inputs, if ever)

Reply:
We're going to suggest this should be a separate PR then that focuses on fixing the CMSIS-DAP implementation's handling of the protocol version and we'll leave this item unresolved until a PR is made but merge this PR anyway.